packages/runtime/container-runtime/src/opCompressor.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { IBatchMessage } from "@fluidframework/container-definitions";
+import { compress } from "lz4js";
+import { BatchMessage } from "./batchManager";
+import { CompressionAlgorithms, ContainerRuntimeMessage } from "./containerRuntime";
+
+/**
+ * Compresses batches of ops. It generates a single compressed op that contains
+ * the contents of each op in the batch. It then submits empty ops for each original
+ * op to reserve sequence numbers.
+ */
+export class OpCompressor {
+    private compressedBatchCount = 0;
+
+    constructor(private readonly logger: ITelemetryLogger) {}
+
+    public compressBatch(batch: BatchMessage[], originalLength: number): IBatchMessage[] {
+        const batchToSend: IBatchMessage[] = [];
+        this.compressedBatchCount++;
+        const batchedContents: ContainerRuntimeMessage[] = [];
+
+        for (const message of batch) {
+            batchedContents.push(message.deserializedContent);
+        }
+
+        const compressionStart = Date.now();
+        const contentsAsBuffer = new TextEncoder().encode(JSON.stringify(batchedContents));
```

Comment:
I believe TextEncoder() is not available in node.js environment. Please use stringToBuffer()

Reply:
Changed to stringToBuffer() in latest commit

Reply:
Hmm, I'm not sure why but it looks like I'm getting and exception from the lz4 library we're using when I use `stringToBuffer`. Are we sure TextEncoder isn't available in Node.js? It looks like it should be [here](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder)

Reply:
Looks like it exists :) But the fact that it crashes seems really suspicious - it's just binary data, it should not crash on any input. Assuming we use same type of buffer